### ISSUE TITLE
Wood does not place right using easy place

### ISSUE THREAD

**Issue by js802025** [maruohon/litematica#319]
**NOTE**:
* Before reporting an issue, please make sure it has not already been fixed in a later version! Also make sure you have the latest version of **both Litematica and malilib!** There are occasionally changes that **will require** both mods to be updated at the same time, or something will break! The very latest development versions are available here: https://masa.dy.fi/mcmods/client_mods/ so you can quickly check if you have the latest builds. The versions on CurseForge are not usually the very latest builds, but important bugfixes will also be uploaded there.
* Please don't post simple questions like "How do I do X?" on this issue tracker. Such help requests should happen on Discord instead: https://discord.gg/VNVG3Wk (or if the previous link is broken: https://discord.gg/scicraft -> `#masas-mods`)
* Don't post issues for mod versions made for Minecraft snapshots on this tracker. The snapshot versions aren't "properly" supported or tested, they are provided as quick ports (when I have the time to make them) in a "might work, mostly" basis, and those issues should rather be handled on Discord as well.

**Minecraft version**
Which Minecraft version are you playing on?
1.16.4
**Mod version and malilib version**
litematica-fabric-1.16.3-1.6
**Description of the bug**
When placing logs/wood with easy place the orientation will be wrong. This is not a problem with the server as it is running fabric with servux installed. Is there a setting I am missing on client/server or is this not possible to solve?

**Screenshots**
![image](https://user-images.githubusercontent.com/30957566/112569309-f2f80380-8db1-11eb-9a73-75594846a0dd.png)

**Comment by maruohon** [comment]
Currently Litematica does not rotate any blocks on its own yet. For single player the rotations will work if you simply have the Tweakeroo mod installed, with default settings (the `clientPlacementRotation` option is what allows it to work, and it's enabled by default). The rotation will get added to Litematica probably in the next round or larger updates.

**Comment by js802025** [comment]
But pistons seem to work with litematica... maybe just luck thanks!

**Comment by maruohon** [comment]
Hmm, I was kinda wondering, wood should actually not work at all. Oh I actually misread or misunderstood your mention of the server. Servux currently doesn't yet have any support for the accurate placement protocol, but it will get it at some point. Currently on a server you need either QuickCarpet or FabricCarpet with CarpetExtra and the `accurateBlockPlacement` rule enabled for any of the rotations to work.

And the protocol currently does not include any blocks that use rotation properties other than a Direction-enum-based property. And logs don't use Direction as they have an axis-based rotation property instead of a 6-way Direction based property.

So basically I just need to add support for a number of these blocks such as logs, and some of the newer blocks like grindstones etc. And that needs to be added both to Litematica, and to the server-side mod handling the requests there. Likely this will not happen for CarpetExtra, but only for Servux when it gets the support.

**Comment by js802025** [comment]
Yea u have carpet too thanks for the clarification

**Comment by ExSalty0** [comment]
> Currently Litematica does not rotate any blocks on its own yet. For single player the rotations will work if you simply have the Tweakeroo mod installed, with default settings (the `clientPlacementRotation` option is what allows it to work, and it's enabled by default). The rotation will get added to Litematica probably in the next round or larger updates.

Sorry for the request but I have done what you said on the server I am hosting myself (Installing fabric carpet + Carpet extra and I turned on accurateBlockPlacement) but it still wont work
I hope you can help with that

**Comment by maruohon** [comment]
@ExSalty0 Try forcing the Generic -> `easyPlaceProtocolVersion` option to `Version 2` in Litematica. If Litematica can't see the `carpet:hello` packet from the server, then it won't know to switch from `Slabs Only` to `Version 2` protocol. I'm not sure what the most common cause is for Litematica to not see that packet, but it seems to be a somewhat common issue. But also note that you shouldn't leave it at `Version 2` on servers that don't have CarpetExtra, or in single player.
